src/hsl.rs

```diff
@@ -13,70 +16,70 @@ use {Color, Rgb, Luma, Xyz, Lab, Lch, Hsv, ColorSpace, Mix, Shade, GetHue, Hue,
 ///
 ///See [HSV](struct.Hsv.html) for a very similar color space, with brightness instead of lightness.
 #[derive(Clone, Copy, Debug, PartialEq)]
-pub struct Hsl {
+pub struct Hsl<T: Float = f32> {
     ///The hue of the color, in degrees. Decides if it's red, blue, purple,
     ///etc.
-	pub hue: RgbHue,
+    pub hue: RgbHue<T>,
```

Comment:
Keep these tab -> space changes. The tabs must have been from my stupid laptop and spaces is correct.